metrics/metrics.go

```diff
@@ -15,21 +15,24 @@ const (
 	activitySuccessCount = "activity_succeeded"
 )
 
-func recordActivityStart(scope tally.Scope, activityType string, scheduledTimeNanos int64) (tally.Scope, tally.Stopwatch) {
-	scope = scope.Tagged(map[string]string{"operation": activityType})
+func recordActivityStart(
+	handler client.MetricsHandler,
+	activityType string,
+	scheduledTimeNanos int64,
+) client.MetricsHandler {
```

Comment:
_Slightly_ confusing to return `metrics.MetricsHandler` here (though reading the code - I get it). Any opinion on returning a `func(error)` that closes over the handler and the startTime?

Reply:
This is samples repo. All code here should be straight forward as much as possible. For example we used to have many helper methods here which I replace with simple copy/paste. 
But in this specific case, I agree. Having something like:
```go
var err error
recordEndFunc := recordActivityStart(...)
defer recordEndFunc(err)
```
looks very natural for go developers. Similar to `ctx, cancel := context.WithCancel(ctx)`.

Reply:
My goal here was to keep as much the same as reasonable to update the SDK. I think a general question of improving this sample or any other is valid, but separate.